jormungandr/src/rest/mod.rs

```diff
@@ -30,8 +30,9 @@ pub async fn start_rest_server(config: Rest, explorer_enabled: bool, context: Co
         .await
         .set_server_stopper(ServerStopper(stopper_tx));
 
-    let api =
-        warp::path!("api" / ..).and(v0::filter(context.clone()).or(v1::filter(context.clone())));
+    let api = warp::path!("api" / ..)
+        .and(v0::filter(context.clone()).or(v1::filter(context.clone())))
+        .with(warp::filters::trace::request());
```

Comment:
Looks like a closure passed to `trace` function would be an improvement because:

* It will allow us to choose the log level.
* Passing B3 information from the headers needs `tracing::Info`.

Reply:
> It will allow us to choose the log level

No. Levels are hardcoded in `warp`. We can only control the verbosity level of the whole span.

If you want to move `warp` traces to another level then it is the time to implement logging by ourselves. Which is not really hard, but is still an effort.

Reply:
We could write our own log filter (later?), shouldn't be difficult.

Reply:
>  info level is correct for a basic request log

We log fragments received from the rest endpoint at the `debug` level, I think it would be a bit strange to have tracing about those requests at a higher level than the transactions themselves

Reply:
> We can only control the verbosity level of the whole span.

I meant exactly that, you can construct a debug span in the closure passed to `trace::trace`.